src/slicer/SlicerArrows.js

```diff
@@ -0,0 +1,187 @@
+      const scalarDirection = (1 / scalar2d) * Math.abs(scalar2d);
+      const scalar3d = distance / Cartesian3.distance(arrowPosition3d, oppositePosition3d) * scalarDirection;
+      const axisVector3d = Cartesian3.subtract(oppositePosition3d, arrowPosition3d, new Cartesian3());
+      const objectMoveVector3d = Cartesian3.multiplyByScalar(axisVector3d, scalar3d, new Cartesian3());
+      const newArrowPosition3d = Cartesian3.add(arrowPosition3d, objectMoveVector3d, new Cartesian3());
+
+      // directly update arrow position if position callback not provided
+      if (!this.positionUpdateCallback) {
+        this.selectedArrow.position = newArrowPosition3d;
+      }
+      if (this.moveCallback) {
+        // calculate move amount (distance with direction)
+        const moveAmount = distance * getDirectionFromPoints(arrowPosition3d, newArrowPosition3d);
+
+        this.moveCallback(side, moveAmount, objectMoveVector3d);
+      }
+
+    } else {
+      this.highlightArrow(movement.endPosition);
+    }
+    this.viewer.scene.requestRender();
+  }
+
+  createMoveArrows() {
+    const arrowEntityTemplate = {
+      model: {
+        minimumPixelSize: 64,
+        scale: 3000,
+        maximumScale: 10000,
+        shadowMode: ShadowMode.DISABLED,
+        colorBlendMode: ColorBlendMode.MIX,
+        color: SLICING_GEOMETRY_COLOR
+      },
+      properties: {}
+    };
+    this.arrows = {};
+    this.arrowsList.forEach(arrow => {
+      const arrowEntityOptions = arrowEntityTemplate;
+      arrowEntityOptions.properties.side = arrow.side;
+      arrowEntityOptions.model.uri = arrow.uri;
+      if (this.positionUpdateCallback) {
+        arrowEntityOptions.position = new CallbackProperty(() => this.positionUpdateCallback(arrow.side), false);
+      } else {
+        arrowEntityOptions.position = arrow.position;
+      }
+      if (arrow.oppositeSide) {
+        arrowEntityOptions.properties.oppositeSide = arrow.oppositeSide;
+      } else if (arrow.oppositePosition) {
+        arrowEntityOptions.properties.oppositePosition = arrow.oppositePosition;
+      }
+      this.arrows[arrow.side] = new Entity(arrowEntityOptions);
+      this.dataSource.entities.add(this.arrows[arrow.side]);
+    });
+  }
+
+  highlightArrow(position) {
+    const pickedObject = this.viewer.scene.pick(position);
+    const isModelPicked = pickedObject && pickedObject.id && pickedObject.id.model;
+    if (isModelPicked && pickedObject.id.properties && pickedObject.id.properties.side) {
+      this.highlightedArrow = pickedObject.id;
+      document.querySelector('.cesium-widget').style.cursor = 'pointer';
+      this.highlightedArrow.model.color = Color.YELLOW;
+    } else {
+      this.unhighlightArrow();
+    }
+  }
+
+  unhighlightArrow() {
+    if (this.highlightedArrow) {
+      this.highlightedArrow.model.color = SLICING_GEOMETRY_COLOR;
+      this.highlightedArrow = undefined;
+      document.querySelector('.cesium-widget').style.cursor = '';
```

Comment:
we should not access the document. If you need to you can certainly do this on the canvas itself.